hail/src/main/scala/is/hail/io/index/IndexWriter.scala

```diff
@@ -78,10 +79,14 @@ object IndexWriter {
     attributes: Map[String, Any] = Map.empty[String, Any]
   ): (FS, String) => IndexWriter = {
     val leafPType = LeafNodeBuilder.typ(keyType, annotationType)
-    val makeLeafEnc = TypedCodecSpec(leafPType, BufferSpec.default).buildEncoder(leafPType)
+    val makeLeafEnc = TypedCodecSpec(EType.defaultFromPType(leafPType),
```

Comment:
can we back these out? It's not a semantic change in the code, right?

Reply:
correct. done

Reply:
Great! Will approve when you push